File: go/PointList/app/pointlist/list.go
package pointlist

import "fmt"

const LIST_LEN int = 100

type List struct {
	arr         [LIST_LEN]Point
	numOfData   int
	curPosition int
}

func (plist *List) ListInit() {
	plist.numOfData = 0
	plist.curPosition = -1
}

func (plist *List) LInsert(data Point) {
	if plist.numOfData >= LIST_LEN {
		fmt.Printf("저장이 불가능 합니다.")
		return
	}

	plist.arr[plist.numOfData] = data
	plist.numOfData += 1
}

func (plist *List) LFirst(pdata *Point) bool {
	if plist.numOfData == 0 {
		return false
	}

	plist.curPosition = 0
	*pdata = plist.arr[0]
	return true
}

func (plist *List) LNext(pdata *Point) bool {
	if plist.curPosition >= plist.numOfData-1 {
		return false
	}

	plist.curPosition += 1
	*pdata = plist.arr[plist.curPosition]
	return true
}

func (plist *List) LRemove() {
	var rpos int = plist.curPosition
	var num int = plist.numOfData

	for i := rpos; i < num-1; i++ {
		plist.arr[i] = plist.arr[i+1]
	}

	plist.numOfData -= 1
	plist.curPosition -= 1
}

func (plist *List) LCount() int {
	return plist.numOfData
}

File: python/PointList/app/pointlist/list.py
from typing import List, Optional
from pydantic import BaseModel, Field

from pointlist.point import Point

class List(BaseModel):
    arr: List[int] = Field(default_factory=list)
    numOfData: int = 0
    curPosition: int = -1

    def LInsert(self, data: Point) -> None:
        self.arr.append(data)
        self.numOfData += 1

    def LFirst(self) -> bool:
        if self.numOfData == 0:
            return False

        self.curPosition = 0
        return True
    
    def LNext(self) -> bool:
        if self.curPosition >= self.numOfData -1:
            return False
        
        self.curPosition += 1
        return True
    
    def LRemove(self):
        rpos: int = self.curPosition
        num: int = self.numOfData

        for i in range(rpos, num-1):
            self.arr[i] = self.arr[i+1]
        
        self.numOfData -= 1
        self.curPosition -= 1
    
    def LCount(self) -> int:
        return self.numOfData